### ISSUE TITLE
More VPC endpoint issues

### ISSUE THREAD

**Issue by sworisbreathing** [spulec/moto#3540]
Piggybacking off of #3524 and localstack/localstack#3310 I've had another attempt at managing VPC endpoints in localstack with terraform, and here are the issues I'm finding (as of moto 1.3.16 or whatever version localstack uses)

* When a VPC endpoint is created and `privateDnsEnabled` was not specified, it defaults to `None` if not specified. This breaks terraform because when it goes to describe the VPC endpoint, `None` is returned for a boolean field, and aws-sdk-go doesn't coerce `None` to `false`.
* I can't fix the VPC endpoint by recreating with `privateDnsEnabled` set explicitly to `false`, because `delete-vpc-endpoints` isn't implemented in moto.

Minimal terraform templates to reproduce (terraform version is 0.14 and localstack is latest version spun up using their sample docker-compose.yml):

```hcl
terraform {
  required_version = "= 0.14.0"
  required_providers {
    aws = "= 3.20"
  }
}

resource "aws_vpc_endpoint" "s3" {
  service_name = "com.amazonaws.ap-southeast-2.s3"
  vpc_id = aws_vpc.main.id
  auto_accept = true
  security_group_ids = [aws_security_group.vpc_endpoints.id]
}

resource "aws_vpc" "main" {
  cidr_block = "192.168.0.0/22"
  instance_tenancy = "default"
  enable_dns_support = true
  enable_dns_hostnames = true
  assign_generated_ipv6_cidr_block = false
}

resource "aws_security_group" "vpc_endpoints" {
  name = "vpc-endpoints"
  vpc_id = aws_vpc.main.id
  ingress {
    protocol = "-1"
    from_port = 0
    to_port = 0
    cidr_blocks = [aws_vpc.main.cidr_block]
  }
  egress {
    protocol = "-1"
    from_port = 0
    to_port = 0
    cidr_blocks = ["0.0.0.0/0"]
  }
}

provider "aws" {
  region  = "ap-southeast-2"
  access_key = "test"
  secret_key = "test"
  skip_credentials_validation = true
  skip_metadata_api_check = true
  skip_requesting_account_id = true
  s3_force_path_style = true
  endpoints {
    ec2 = "http://localhost:4566"
    # ... all endpoints pointing to http://localhost:4566
  }
}
````

**Comment by bblommers** [comment]
Thanks for raising this @sworisbreathing! Marking it as a bug.

**Comment by apregoe** [comment]
Any new updates on this?

**Comment by bblommers** [comment]
The `delete_vpc_endpoints` feature has now been implemented by #4169, and should be available as of moto >= 2.2.3.
I'll close this for now, but let us know if we've missed anything.
